pkg/logging/logging.go

```
@@ -0,0 +1,63 @@
+func GetLogger() *Logger {
+	once.Do(func() {
+		logger = &Logger{slog.New(slog.NewJSONHandler(logWriter(), &slog.HandlerOptions{
+			Level: slog.LevelInfo,
+		}))}
+	})
+	return logger
+}
+
+func logWriter() io.Writer {
+	if path := os.Getenv(constants.LogFileEnvVar); path != "" {
+		f, err := os.OpenFile(path, os.O_APPEND|os.O_WRONLY, 0)
```

golangci-lint (2.12.2)

[high] G703: Path traversal via taint analysis

(gosec)